src/R/berekening_hulpvariabelen.R

```diff
@@ -239,4 +239,54 @@ bereken_vvi <- function(point,
     output_type = output_type)
 }
 
+add_visibility_to_frame <- function(punten_sf,
+                                    gebied,
+                                    name,
+                                    resolution,
+                                    spacing,
+                                    dist = 300,
+                                    obs_height = 1.7,
+                                    dsm_offset = 0) {
+
+  filename_dsm <- paste0("dhmvii_dsm_1m_", name, ".tif")
+  file_dsm <- file.path(mbag_dir, "data", "dem", filename_dsm)
+
+  if (file.exists(file_dsm)) {
+    dsm <- terra::rast(file_dsm)
+    crs(dsm) <- "epsg:31370"
+  } else {
+    dsm <- terra::rast("S:/Vlaanderen/Hoogte/DHMVII/DHMVIIDSMRAS1m.tif") %>%
+      terra::crop(gebied, filename = file_dsm)
+    crs(dsm) <- "epsg:31370"
+  }
+
+  filename_dtm <- paste0("dhmvii_dtm_1m_", name, ".tif")
+  file_dtm <- file.path(mbag_dir, "data", "dem", filename_dtm)
+
+  if (file.exists(file_dtm)) {
+    dtm <- terra::rast(file_dtm)
+    crs(dtm) <- "epsg:31370"
+  } else {
+    dtm <- terra::rast("S:/Vlaanderen/Hoogte/DHMVII/DHMVIIDTMRAS1m.tif") %>%
+      terra::crop(gebied, filename = file_dtm)
+    crs(dtm) <- "epsg:31370"
+  }
+
+  vvi_dm <- vvi_from_sf(
+    observer = punten_sf %>% st_buffer(dist = 25),
+    spacing = spacing,
+    cores = 1,
+    progress = TRUE,
+    max_distance = dist,
+    dsm_rast = dsm + dsm_offset,
```

Comment:
Die `dsm_offset` mag je verwijderen, ook in het functie-argument. Dat was een idee waarmee ik speelde, maar dat niet doet wat ik wou dat het deed (mijn doel was: wat is de zichtbaarheid van een object x meter boven maaiveld).